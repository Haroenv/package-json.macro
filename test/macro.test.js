const pluginTester = require('babel-plugin-tester');
const plugin = require('babel-plugin-macros');

pluginTester({
  plugin,
  snapshot: true,
  babelOptions: { filename: __filename },
  tests: [
    `
      import packageJson from '../src/macro';

      // simple key
      const description = packageJson('description');
    `,
    `
      import packageJson from '../src/macro';

      // object
      const jest = packageJson('jest');
    `,
    `
      import packageJson from '../src/macro';

      // deep object
      const jestTestEnvironment = packageJson('jest.testEnvironment');
    `,
    `
      import packageJson from '../src/macro';

      // array
      const forTestingArray = packageJson('_forTesting.array');
    `,
    `
      import packageJson from '../src/macro';

      // array (deep)
      const forTestingArrayOne = packageJson('_forTesting.array[1]');
      const forTestingArrayOneKey = packageJson('_forTesting.array[1].object');
      const forTestingArrayOneKeyFake = packageJson('_forTesting.array[1].objectFake');
    `,
    `
      import packageJson from '../src/macro';

      // undefined when doesn't exist
      packageJson('_blibli_doesnt_exist');

      // null when actually in data
      packageJson('_forTesting.isNull');
    `,
    `
    import packageJson from '../src/macro'

    // kitchensink
    const version = packageJson('version')
    const name = packageJson('name')
    const object = packageJson('_forTesting.array[1].object')
    `,
  ],
});
