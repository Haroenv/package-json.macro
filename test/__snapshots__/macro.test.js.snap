// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros 1. macros: 1. macros 1`] = `
"
import packageJson from '../src/macro';

// simple key
const description = packageJson('description');

      ↓ ↓ ↓ ↓ ↓ ↓

// simple key
const description = 'babel macro for reading only part of package.json';
"
`;

exports[`macros 2. macros: 2. macros 1`] = `
"
import packageJson from '../src/macro';

// object
const jest = packageJson('jest');

      ↓ ↓ ↓ ↓ ↓ ↓

// object
const jest = {
  testEnvironment: 'node'
};
"
`;

exports[`macros 3. macros: 3. macros 1`] = `
"
import packageJson from '../src/macro';

// deep object
const jestTestEnvironment = packageJson('jest.testEnvironment');

      ↓ ↓ ↓ ↓ ↓ ↓

// deep object
const jestTestEnvironment = 'node';
"
`;

exports[`macros 4. macros: 4. macros 1`] = `
"
import packageJson from '../src/macro';

// array
const forTestingArray = packageJson('_forTesting.array');

      ↓ ↓ ↓ ↓ ↓ ↓

// array
const forTestingArray = ['hi', {
  object: true
}];
"
`;

exports[`macros 5. macros: 5. macros 1`] = `
"
import packageJson from '../src/macro';

// array (deep)
const forTestingArrayOne = packageJson('_forTesting.array[1]');
const forTestingArrayOneKey = packageJson('_forTesting.array[1].object');
const forTestingArrayOneKeyFake = packageJson('_forTesting.array[1].objectFake');

      ↓ ↓ ↓ ↓ ↓ ↓

// array (deep)
const forTestingArrayOne = {
  object: true
};
const forTestingArrayOneKey = true;
const forTestingArrayOneKeyFake = undefined;
"
`;

exports[`macros 6. macros: 6. macros 1`] = `
"
import packageJson from '../src/macro';

// undefined when doesn't exist
packageJson('_blibli_doesnt_exist');

// null when actually in data
packageJson('_forTesting.isNull');

      ↓ ↓ ↓ ↓ ↓ ↓

// undefined when doesn't exist
undefined;

// null when actually in data
null;
"
`;
